### ISSUE TITLE
💅  lint/suspicious/noArrayIndexKey false negative when re-ordering value and index in template string

### ISSUE THREAD

**Issue by giavinh79** [biomejs/biome#8812]
### Environment information

<details>

```bash
CLI:
  Version:                      2.3.10
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.20.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset
```

</details>

### Rule name

lint/suspicious/noArrayIndexKey

### Playground link

https://biomejs.dev/playground/?lintRules=noArrayIndexKey&code=YwBvAG4AcwB0ACAAUgBlAG4AZABlAHIAQQByAHIAYQB5ACAAPQAgACgAKQAgAD0APgAgAHsACgAgACAAYwBvAG4AcwB0ACAAYQByAHIAIAA9ACAAWwAxACwAMgAsADMAXQA7AAoAIAAgAHIAZQB0AHUAcgBuACAAYQByAHIALgBtAGEAcAAoACgAaQB0AGUAbQAsACAAaQBuAGQAZQB4ACkAIAA9AD4AIAB7AAoAIAAgACAAIAAvAC8AIAB0AGgAcgBvAHcAcwAgAGUAcgByAG8AcgAgAGEAcwAgAGUAeABwAGUAYwB0AGUAZAAKACAAIAAgACAAcgBlAHQAdQByAG4AIAA8AGQAaQB2ACAAawBlAHkAPQB7AGAAJAB7AGkAdABlAG0AfQAtACQAewBpAG4AZABlAHgAfQBgAH0APgB7AGkAdABlAG0AfQA8AC8AZABpAHYAPgA7AAoAIAAgAH0AKQAKAH0ACgAKAGMAbwBuAHMAdAAgAFIAZQBuAGQAZQByAEEAcgByAGEAeQAyACAAPQAgACgAKQAgAD0APgAgAHsACgAgACAAYwBvAG4AcwB0ACAAYQByAHIAIAA9ACAAWwAxACwAMgAsADMAXQA7AAoAIAAgAHIAZQB0AHUAcgBuACAAYQByAHIALgBtAGEAcAAoACgAaQB0AGUAbQAsACAAaQBuAGQAZQB4ACkAIAA9AD4AIAB7AAoAIAAgACAAIAAvAC8AIABuAG8AIABlAHIAcgBvAHIAIAB0AGgAcgBvAHcAbgAsACAAYgB1AHQAIABlAHgAcABlAGMAdABlAGQAIABhAG4AIABlAHIAcgBvAHIACgAgACAAIAAgAHIAZQB0AHUAcgBuACAAPABkAGkAdgAgAGsAZQB5AD0AewBgACQAewBpAG4AZABlAHgAfQAtACQAewBpAHQAZQBtAH0AYAB9AD4AewBpAHQAZQBtAH0APAAvAGQAaQB2AD4AOwAKACAAIAB9ACkACgB9AA%3D%3D

### Expected result

Good scenario - correctly reports an error as the `key` is relying on the `index` of the array:
```JSX
<div key={`${item}-${index}`}>
``` 

Edge case - when rearranging the order of the values in the key, we should still get an error but we don't:
```JSX
<div key={`${index}-${item}`}>
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by giavinh79** [comment]
I've also noticed that object keys (even with `index` being the last variable within the template string) also break the lint rule i.e. `${item.title}-${index}`

**Comment by LouisLau-art** [comment]
Opened PR #8968 to fix this: noArrayIndexKey now flags the array index even when it's not the last identifier in a template/binary expression (and when templates include member expressions like item.title). https://github.com/biomejs/biome/pull/8968
